### PR TITLE
KSQL-12238: backport vertx 4.3 changes to 7.2.x (#37)

### PR DESCRIPTION
**Pull request by kagarwal06**
* KSQL-12238: backport vertx 4.3 changes to 7.2.x

* KSQL-12238: uncomment tests

Commits backported:

1. https://github.com/confluentinc/ksql/pull/9427
2. https://github.com/confluentinc/ksql/pull/9446
3. https://github.com/confluentinc/ksql/pull/9540

### Description 
_What behavior do you want to change, why, how does your patch achieve the changes?_

### Testing done 
_Describe the testing strategy. Unit and integration tests are expected for any behavior changes._

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")
- [ ] Do these changes have compatibility implications for rollback? If so, ensure that the ksql [command version](https://github.com/confluentinc/ksql/blob/master/ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/computation/Command.java#L41) is bumped.

### CODE REVIEW

File: ksqldb-rest-app/src/test/java/io/confluent/ksql/api/CorsTest.java [hunk 1]
```diff
@@ -251,7 +251,7 @@ private void shouldAcceptPreflightRequest(final String origin, final String allo
         MultiMap.caseInsensitiveMultiMap().add("origin", origin)
             .add(ACCESS_CONTROL_REQUEST_METHOD_HEADER, "POST"));
 
-    assertThat(response.statusCode(), is(200));
+    assertThat(response.statusCode(), is(204));
```
Comment (hrishabhg) [inline ksqldb-rest-app/src/test/java/io/confluent/ksql/api/CorsTest.java:254]: Though the change is matching. Not sure it is standard practice for `OPTIONS` call. [fyi]

File: ksqldb-rest-app/src/main/java/io/confluent/ksql/api/server/QuerySubscriber.java [hunk 2]
```diff
@@ -106,7 +106,7 @@ public void handleError(final Throwable t) {
     }
     final KsqlErrorMessage errorResponse = new KsqlErrorMessage(
         ERROR_CODE_SERVER_ERROR, stringBuilder.toString());
-    log.error("Error in processing query {}", stringBuilder);
+    log.error("Error in processing query {}", stringBuilder, t);
```
Comment (hrishabhg) [inline ksqldb-rest-app/src/main/java/io/confluent/ksql/api/server/QuerySubscriber.java:109]: We can revert this change. Just reducing the pieces not required to avoid regression.

File: ksqldb-rest-app/src/main/java/io/confluent/ksql/api/server/AuthHandlers.java [hunk 3]
```diff
@@ -52,50 +57,72 @@ static void setupAuthHandlers(final Server server, final Router router,
 
     systemAuthenticationHandler.ifPresent(handler -> router.route().handler(handler));
 
-    if (jaasAuthHandler.isPresent() || authenticationPlugin.isPresent()) {
-      router.route().handler(AuthHandlers::pauseHandler);
+    if (jaas.isPresent() || authenticationPlugin.isPresent()) {
```
Comment (hrishabhg) [inline ksqldb-rest-app/src/main/java/io/confluent/ksql/api/server/AuthHandlers.java:60]: these changes are not matching with 7.3.x.

File: ksqldb-api-client/src/main/java/io/confluent/ksql/api/client/impl/QueryResponseHandler.java [hunk 4]
```diff
@@ -62,7 +59,8 @@ private void handleArgs(final Buffer buff) {
 
     final QueryResponseMetadata queryResponseMetadata;
     try {
-      queryResponseMetadata = JSON_MAPPER.readValue(buff.getBytes(), QueryResponseMetadata.class);
+      queryResponseMetadata = ApiJsonMapper.INSTANCE.get()
```
Comment (hrishabhg) [inline ksqldb-api-client/src/main/java/io/confluent/ksql/api/client/impl/QueryResponseHandler.java:62]: this is not matching with 7.3.x . Either we can keep the older version with JsonMapper or we can handle it the way 7.3.x does.

**Review by hrishabhg — CHANGES_REQUESTED** [review]
1. Mostly okay.
2. 1 critical change - handling media type
3. 2-3 files are not matching with 7.3.x. Re-review them once. (Changes might be okay)

File: ksqldb-engine/src/main/java/io/confluent/ksql/execution/scalablepush/PushRouting.java [hunk 5]
```diff
@@ -291,8 +291,8 @@ static CompletableFuture<RoutingResult> executeOrRouteQuery(
       final String thisHostName
   ) {
     if (node.isLocal()) {
-      LOG.info("Query {} id {} executed locally at host {} at timestamp {}.",
-          statement.getMaskedStatementText(), pushPhysicalPlanManager.getQueryId(), node.location(),
+      LOG.info("Query with id {} executed locally at host {} at timestamp {}.",
```
[the remark this answers is not in the document]
Comment (kagarwal06) [reply ksqldb-engine/src/main/java/io/confluent/ksql/execution/scalablepush/PushRouting.java]: Done

File: ksqldb-engine/src/main/java/io/confluent/ksql/execution/pull/StreamedRowTranslator.java [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+      final LogicalSchema expected,
+      final LogicalSchema actual,
+      final Optional<KsqlHostInfoEntity> sourceHost) {
+    if (!actual.equals(expected)) {
+      throw new KsqlException(
+          String.format("Schemas %s from host %s differs from schema %s",
+              actual, sourceHost .map(KsqlHostInfoEntity::getHost).orElse("unknown"), expected));
+    }
+  }
+
```
[the remark this answers is not in the document]
Comment (kagarwal06) [reply ksqldb-engine/src/main/java/io/confluent/ksql/execution/pull/StreamedRowTranslator.java]: Done

File: ksqldb-api-client/src/main/java/io/confluent/ksql/api/client/impl/ClientImpl.java [hunk 7]
```diff
@@ -604,26 +605,45 @@ private <T extends CompletableFuture<?>> void makeRequest(
       final Handler<HttpClientResponse> responseHandler,
       final boolean endRequest,
       final HttpMethod method) {
-    HttpClientRequest request = httpClient.request(method,
-        serverSocketAddress, clientOptions.getPort(), clientOptions.getHost(),
-        path,
-        responseHandler)
-        .exceptionHandler(cf::completeExceptionally);
```
[the remark this answers is not in the document]
Comment (kagarwal06) [reply ksqldb-api-client/src/main/java/io/confluent/ksql/api/client/impl/ClientImpl.java:611]: Discussed offline

File: ksqldb-rest-app/src/main/java/io/confluent/ksql/api/server/QuerySubscriber.java (hunk 2 above)
Comment (kagarwal06) [reply ksqldb-rest-app/src/main/java/io/confluent/ksql/api/server/QuerySubscriber.java]: Done

File: ksqldb-rest-app/src/main/java/io/confluent/ksql/api/server/AuthHandlers.java (hunk 3 above)
Comment (kagarwal06) [reply ksqldb-rest-app/src/main/java/io/confluent/ksql/api/server/AuthHandlers.java:60]: Thanks for pointing this out. I see there was one more commit done post the main changes. I have added that commit as well(https://github.com/confluentinc/ksql/pull/9446)

File: ksqldb-api-client/src/main/java/io/confluent/ksql/api/client/impl/QueryResponseHandler.java (hunk 4 above)
Comment (kagarwal06) [reply ksqldb-api-client/src/main/java/io/confluent/ksql/api/client/impl/QueryResponseHandler.java:62]: I checked. Internally both are calling same APIs. Will keep it as is.

**Review by hrishabhg — APPROVED** [review]
